File: API/DTOs/report/QRCodeWIPReport/QRCodeDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.DTOs.report.QRCodeWIPReport
{
    public class CheckSumMissHeaderDTO
    {
        [StringLength(6)]
        public string yymmdd{get;set;}
        public decimal? Sumpackage { get; set; }
        public decimal? Sumpairs { get; set; }
        public decimal? SumSortPack { get; set; }
        public decimal? SumSortPairs { get; set; }
        public decimal? SumlossPairs { get; set; }
        public decimal? SumStoragePack { get; set; }
        public decimal? SumStoragePairs { get; set; }
        public decimal? MissPackage { get; set; }
        public decimal? MissPairs { get; set; }
        public decimal? Proportion { get; set; }
    }
    public class CheckSumMissDetailDTO{
        [StringLength(6)]
        public string yymmdd {get;set;}
        public string Manuf {get;set;}
        public string Type {get;set;}
        public string Manno {get;set;}
        public string Purno {get;set;}
        public string prtno {get;set;}
        public string wkshno {get;set;}
        public string QRCodeID {get;set;}
        public string rmodel {get;set;}
        public string tolcls {get;set;}
        public string bitnbr {get;set;}
        public DateTime? eta {get;set;}
        public string empno {get;set;}
        public short? serial {get;set;}
        public string size {get;set;}
        public decimal? qty {get;set;}
        public string Crusr {get;set;}
        public DateTime? Crday {get;set;}
        public string CrdaY_str { get => Crday.HasValue ? Crday.Value.ToString("yyyy-MM-dd HH:mm:ss") : ""; }
        public string softFlag {get;set;}
        public string storageFlag {get;set;}
    }

}

File: SPA/src/app/_core/models/qRCodeDTO.ts
export interface CheckSumMissHeaderDTO {
    yymmdd: string;
    sumPackage: number | null;
    sumPairs: number | null;
    sumSortPack: number | null;
    sumSortPairs: number | null;
    sumLossPairs: number | null;
    sumStoragePack: number | null;
    sumStoragePairs: number | null;
    missPackage: number | null;
    missPairs: number | null;
    proPortion: number | null;
}

export interface CheckSumMissDetailDTO {
    yymmdd: string;
    manuf: string;
    type: string;
    manno: string;
    purno: string;
    prtno: string;
    wkshno: string;
    qRCodeID: string;
    rmodel: string;
    tolcls: string;
    bitnbr: string;
    eta: string | null;
    empno: string;
    serial: number | null;
    size: string;
    qty: number | null;
    crusr: string;
    crday: string | null;
    softFlag: string;
    storageFlag: string;
}